### PR TITLE
Add UI for setting an area on entity level

### PR DESCRIPTION
**Pull request by bramkragten**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/entities/entity-registry-basic-editor.ts [hunk 1]
```diff
@@ -105,6 +141,12 @@ export class HaEntityRegistryBasicEditor extends LitElement {
         .invalid=${invalidDomainUpdate}
         .disabled=${this._submitting}
       ></paper-input>
+      <ha-area-picker
```
Comment (balloob) [inline src/panels/config/entities/entity-registry-basic-editor.ts:144]: I think that we should only show it if there is no device linked to this entity.

Comment (balloob) [reply src/panels/config/entities/entity-registry-basic-editor.ts:144]: Or maybe we should have it control the device area if it has a linked device.

Comment (balloob) [reply src/panels/config/entities/entity-registry-basic-editor.ts:144]: Setting the area of an entity that belongs to a device should be the exception.

Comment (bramkragten) [reply src/panels/config/entities/entity-registry-basic-editor.ts:144]: But then how would you handle the exception?

Comment (balloob) [reply src/panels/config/entities/entity-registry-basic-editor.ts:144]: Inside an advanced accordion, with a checkbox that is checked by default "Use Device Area", uncheck it and then the dropdown enables, you can pick it.

Comment (bramkragten) [reply src/panels/config/entities/entity-registry-basic-editor.ts:144]: I don't think the checkbox works, what would you expect to happen when you uncheck the `Use Device Area`, but don't select an area?

Comment (bramkragten) [reply src/panels/config/entities/entity-registry-basic-editor.ts:144]: Maybe like this:
area-device-entity

**Review by balloob — APPROVED** [review]

**Pull request merged**